source/MaterialXGraphEditor/Graph.cpp

```diff
@@ -811,29 +814,24 @@ void Graph::updateMaterials(mx::InputPtr input, mx::ValuePtr value)
     {
         if (!input)
         {
-            mx::ElementPtr elem = nullptr;
-            {
-                elem = _graphDoc->getDescendant(renderablePath);
-            }
+            mx::ElementPtr elem = _graphDoc->getDescendant(renderablePath);
             mx::TypedElementPtr typedElem = elem ? elem->asA<mx::TypedElement>() : nullptr;
             _renderer->updateMaterials(typedElem);
         }
         else
         {
-            std::string name = input->getNamePath();
             // need to use exact interface name in order for input
+            std::string name = input->getNamePath();
             mx::InputPtr interfaceInput = findInput(input, input->getName());
             if (interfaceInput)
             {
                 name = interfaceInput->getNamePath();
             }
-            // Note that if there is a topogical change due to
```

Comment:
This is too late to check for topology changes. Done on value change instead before calling this code. Also makes this method only either do a value or rebuild without any hidden logic.